### PR TITLE
Base Network handler, model classes

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75277: Content](https://edu-redmine.sun-asterisk.vn/issues/75277)
## WHAT
- Create model classes
- Annotations for Api marking
- Base Network handler using Java/Kotlin Reflection to handle Api Marking
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools">
 
+    <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
+
+
```
Comment (daolq-2712) [inline app/src/main/AndroidManifest.xml]:
```suggestion
```

**Review by ghost — APPROVED** [review]

File: app/src/main/java/com/example/travenor/data/model/place/PlacePhoto.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    @SerializedName("small") val small: Image,
+    @SerializedName("medium") val medium: Image,
+    @SerializedName("large") val large: Image,
+    @SerializedName("original") val original: Image
+)
+
+data class Image(
+    @SerializedName("height") val height: Int,
+    @SerializedName("width") val width: Int,
+    @SerializedName("url") val url: String
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/data/model/place/PlacePhoto.kt]: tạo ra 3 class khác nhau đi em

File: app/src/main/java/com/example/travenor/data/model/review/PlaceTip.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.example.travenor.data.model.review
+
+import com.google.gson.annotations.SerializedName
+
+data class PlaceTip(
+    @SerializedName("id") val id: String,
+    @SerializedName("text") val text: String,
+    @SerializedName("agree_count") val agreeCount: Int,
+    @SerializedName("disagree_count") val disagreeCount: Int
+
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/data/model/review/PlaceTip.kt]:
```suggestion
```

File: app/src/main/java/com/example/travenor/data/model/review/Review.kt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import com.google.gson.annotations.SerializedName
+
+data class Review(
+    @SerializedName("id") val id: String,
+    @SerializedName("location_id") val locationId: String,
+    @SerializedName("rating") val rating: Float,
+    @SerializedName("text") val text: String,
+    @SerializedName("title") val title: String,
+    @SerializedName("published_date") val date: String
+
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/data/model/review/Review.kt]: .

File: app/src/main/java/com/example/travenor/data/model/search/LocationSearch.kt [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.example.travenor.data.model.place.Address
+import com.google.gson.annotations.SerializedName
+
+data class LocationSearch(
+    @SerializedName("name") val name: String,
+    @SerializedName("location_id") val locationId: String,
+    @SerializedName("distance") val distance: String,
+    @SerializedName("address_obj") val addressObj: Address
+
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/data/model/search/LocationSearch.kt]: .

File: app/src/main/java/com/example/travenor/screen/main/MainActivity.kt [hunk 6]
```diff
@@ -13,26 +13,8 @@ class MainActivity : BaseActivity() {
         return mBinding.root
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
```
Comment (daolq-2712) [inline app/src/main/java/com/example/travenor/screen/main/MainActivity.kt]:
```suggestion
```

**Review by daolq-2712 — CHANGES_REQUESTED** [review]

**Review by daolq-2712 — APPROVED** [review]
OK!

**Pull request merged**